### DISCUSSION TITLE
CrudField JavaScript Library (question about work )

### DISCUSSION THREAD

**Discussion by guleswine** [Laravel-Backpack/community-forum#251] [Q&A (Help)] [answered]
After reading the documentation, I decided to test the work of subfields
![image](https://user-images.githubusercontent.com/23197378/173246788-679cbcaf-b403-4663-8ab7-63d12b2a0dcd.png)

and expected the following code to work for hiding and showing the required field
but as a result, this code simply hides the main field  (instead of subfields inside)
```
crud.field('phrases').subfield('event_type').onChange(function(field) {
    if (field.value == 'advanced_function') {
        crud.field('phrases').subfield('advanced_function_id', field.rowNumber).show();
        crud.field('phrases').subfield('intent_id', field.rowNumbe).hide();
    } else if (field.value == 'intent') {
        crud.field('phrases').subfield('intent_id', field.rowNumbe).show();
        crud.field('phrases').subfield('advanced_function_id', field.rowNumbe).hide();
    }

}).change();
```
I looked at how the code in the demo works, and as a result I got such a working code
```
crud.field('phrases').subfield('event_type').onChange(function(field) {
    crud.field('phrases').subfield('advanced_function_id', field.rowNumber).show(field.value == 'advanced_function');
    crud.field('phrases').subfield('intent_id', field.rowNumber).show(field.value == 'intent');
}).change();
```
isn't it expected that this code should be equivalent to each other in my case, and the top option should also be workable?

**Comment by pxpm** [comment, marked as answer]
Hello @guleswine 

It's missing an "r" in `rowNumber`. You have `rowNumbe` in your first example.

I think that's what's broke because from my understanding both examples should work. 

Let me know,
Cheers

**Reply by guleswine** [reply]
Oh, really, I didn’t even notice, apparently I had disabled parser  for JS in phpstorm, and the browser console didn’t show any errors either.

**Reply by guleswine** [reply]
turned on the js parser, the problem immediately became visible, the first option is really working

**Reply by pxpm** [reply]
Thanks for confirming @guleswine !

Hope you are enjoying working with it 👍 Let me know if you find some bug or implementation issue.

Cheers
